bookstore-front/src/core/Checkout.js

```diff
@@ -15,17 +39,83 @@ const Checkout = ({ products }) => {
 
     const showCheckout = () => {
         return isAuthenticated() ? (
-            <button className='btn btn-success'>Checkout</button>
+            <div>{showDropIn()}</div>
         ) : (
                 <Link to='/signin'>
                     <button className="btn btn-primary">Sign in to checkout</button>
                 </Link>
             )
     }
 
+    const buy = () => {
+        setData({ loading: true })
+        //send data to backend
+        let nonce;
+        const getNonce = data.instance.requestPaymentMethod()
+            .then(data => {
+                nonce = data.nonce;
```

Comment:
You previously declared `let nonce` on line 53, but you're only using it inside this `.then()`, so you should declare it on line 56 and use `const`.

Reply:
will do